### ISSUE TITLE
Prefer smaller/larger releases

### ISSUE THREAD

**Issue by markus101** [Sonarr/Sonarr#724]
When nzbs have par files they are larger, currently we're grabbing the smallest nzb in that quality, when multiple releases come out with the same, some with par2 files, others without, we're grabbing the smallest release and missing out on par2 files. I think we need to be smarter about which we grab, not the largest (weed out the overly large files), but the smallest probably isn't the most preferred.

**Comment by otoolejp** [comment]
It might be easier (code wise) to just grab the largest, since people can already determine for themselves with quality definitions what is considered an overly large file.

**Comment by spooky69** [comment]
+1

**Comment by Taloth** [comment]
Like over a year ago I did propose a 'preferred' size in addition to the existing min/max.

**Comment by MicahZoltu** [comment]
As a user, I would like to be able to tell Sonarr "Delay downloading for 6 hours after the first episode airs, and then download the smallest release available that matches X filter".

For some things, the quality doesn't bother me, especially if I'm going to be watching it on a mobile device anyway.  Since I'm on a metered internet connection I would like to avoid fetching a 2GB file for a 30 minute show that I'll end up watching on a tiny screen.  However, for some shows the only thing available is a large file and I would rather spend the bandwidth to get that then have nothing at all.

At the moment, I find myself spending a lot of time manually downloading files because I have setup my download profiles to target small encodings.  Sometimes (fairly often) a show won't have an encoding available at the target size, which means I have to go in and manually initiate a download.

**Comment by sonuyos** [comment]
Any update on this? its been 5yrs, lol.

**Comment by dippynark** [comment]
Also surprised this is still not implemented, all that would be required is a switch on the function used to compare size (https://github.com/Sonarr/Sonarr/blob/c75c54688883461d0e5275f349fba957d992f765/src/NzbDrone.Core/DecisionEngine/DownloadDecisionComparer.cs#L191) -- I am maintaining my own fork just for that option

**Comment by markus101** [comment]
Sounds like a PR should be opened so it can be reviewed and implemented.

**Comment by Qstick** [comment]
We've implemented preferred size for quality settings on Radarr v3 which also would be a fairly easy pull over. It solves this as @Taloth mentions above

**Comment by bakerboy448** [comment]
@Qstick looks like that was this PR https://github.com/Radarr/Radarr/pull/4190 ?

**Comment by WernerCD** [comment]
Answers like this got me to my solution: I'm willing to download 1080's and 720's... but prefer 480s... so I got something similar where it goes:

480
DVD
SDTV <= Cutoff
720
1080

**Comment by Ferk** [comment]
> > It might be easier (code wise) to just grab the largest, since people can already determine for themselves with quality definitions what is considered an overly large file.

But I would prefer the smaller files, not larger. And if I limit the quality definitions I just get cut off completelly if there are no results within the threshold. It's a preference, not a hard pass.

I actually might prefer a 1080p video with more compression artifacts than a 720 video with less compression. At the end of the day I'm more concerned about the size.

**Comment by cjmanca** [comment]
Is anyone working on backporting this from Radarr? If not, I'll do up a pull request soon

**Comment by Qstick** [comment]
> Is anyone working on backporting this from Radarr? If not, I'll do up a pull request soon

Nope, all yours

**Comment by Killerherts** [comment]
> Is anyone working on backporting this from Radarr? If not, I'll do up a pull request soon

Are you working on this? I don't want to look into it if it's already done.

**Comment by cjmanca** [comment]
I haven't yet. I was waiting on my previous PR to be merged before doing a new one, which took longer than expected, and then I got distracted by other things. You're welcome to if you'd like, otherwise I'll probably get back to this eventually

**Comment by spikysimonw** [comment]
@cjmanca @Killerherts look forward to either of your work on this coming to fruition, thanks in anticipation.
